### PR TITLE
Convert ReversibleTransactionExecutor to a component

### PR DESCRIPTION
**Pull request by tinchou**
This is (yet again) in preparation for `eth_call` with `pending` parameter.

### CODE REVIEW

**Comment by rskops** [comment]
SonarQube analysis reported 4 issues

1. ![MAJOR][MAJOR] [ReversibleTransactionExecutor.java#L81](https://github.com/rsksmart/rskj/blob/ec9c383a5e4b6ffc388950b5fdb3a9ca75a49b7e/rskj-core/src/main/java/co/rsk/core/ReversibleTransactionExecutor.java#L81): Method has 8 parameters, which is greater than 7 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00107)
1. ![MAJOR][MAJOR] [RskImpl.java#L39](https://github.com/rsksmart/rskj/blob/ec9c383a5e4b6ffc388950b5fdb3a9ca75a49b7e/rskj-core/src/main/java/co/rsk/core/RskImpl.java#L39): Constructor has 8 parameters, which is greater than 7 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00107)
1. ![MAJOR][MAJOR] [EthereumImpl.java#L85](https://github.com/rsksmart/rskj/blob/ec9c383a5e4b6ffc388950b5fdb3a9ca75a49b7e/rskj-core/src/main/java/org/ethereum/facade/EthereumImpl.java#L85): ExecutorService org.ethereum.facade.EthereumImpl.peerServiceExecutor is replaced with another ExecutorService without being shutdown, potentially preventing the entire JVM from shutting down [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3AHES_EXECUTOR_OVERWRITTEN_WITHOUT_SHUTDOWN)
1. ![MAJOR][MAJOR] [EthereumImpl.java#L156](https://github.com/rsksmart/rskj/blob/ec9c383a5e4b6ffc388950b5fdb3a9ca75a49b7e/rskj-core/src/main/java/org/ethereum/facade/EthereumImpl.java#L156): Non derivable method org.ethereum.facade.EthereumImpl$1.adapt(List) declares throwing an exception that isn't thrown [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ABED_BOGUS_EXCEPTION_DECLARATION)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: rskj-core/src/main/java/co/rsk/core/ReversibleTransactionExecutor.java [hunk 1]
```diff
@@ -28,63 +28,107 @@
...
 
-    public static TransactionExecutor executeTransaction(RskSystemProperties config,
-                                                         Repository track,
-                                                         BlockStore blockStore,
-                                                         ReceiptStore receiptStore,
-                                                         ProgramInvokeFactory programInvokeFactory,
-                                                         Block executionBlock,
-                                                         RskAddress coinbase,
-                                                         Web3.CallArguments args) {
+    public ProgramResult executeTransaction(
```
Comment (lsebrie) [inline rskj-core/src/main/java/co/rsk/core/ReversibleTransactionExecutor.java]: I would remove this method from here and unwrap params into EthModule

Comment (tinchou) [reply rskj-core/src/main/java/co/rsk/core/ReversibleTransactionExecutor.java]: Done. I ended up moving `callConstant` to `EthModule` too:

* https://github.com/rsksmart/rskj/pull/489/commits/da77fb31684ba28b3d07fa8e5c1bf58548d7a3f0
* https://github.com/rsksmart/rskj/pull/489/commits/e1e0a40121eb30e99d47df989ec1bdeb62520e04

**Comment by rskops** [comment]
SonarQube analysis reported 15 issues
* ![CRITICAL][CRITICAL] 4 critical
* ![MAJOR][MAJOR] 6 major
* ![MINOR][MINOR] 4 minor
* ![INFO][INFO] 1 info

#### Top 10 issues

1. ![CRITICAL][CRITICAL] [ReversibleTransactionExecutor.java#L29](https://github.com/rsksmart/rskj/blob/e1e0a40121eb30e99d47df989ec1bdeb62520e04/rskj-core/src/main/java/co/rsk/core/ReversibleTransactionExecutor.java#L29): Unused import - org.ethereum.rpc.Web3. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=checkstyle%3Acom.puppycrawl.tools.checkstyle.checks.imports.UnusedImportsCheck)
1. ![CRITICAL][CRITICAL] [ReversibleTransactionExecutor.java#L30](https://github.com/rsksmart/rskj/blob/e1e0a40121eb30e99d47df989ec1bdeb62520e04/rskj-core/src/main/java/co/rsk/core/ReversibleTransactionExecutor.java#L30): Unused import - org.ethereum.rpc.converters.CallArgumentsToByteArray. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=checkstyle%3Acom.puppycrawl.tools.checkstyle.checks.imports.UnusedImportsCheck)
1. ![CRITICAL][CRITICAL] [Ethereum.java#L27](https://github.com/rsksmart/rskj/blob/e1e0a40121eb30e99d47df989ec1bdeb62520e04/rskj-core/src/main/java/org/ethereum/facade/Ethereum.java#L27): Unused import - org.ethereum.rpc.Web3. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=checkstyle%3Acom.puppycrawl.tools.checkstyle.checks.imports.UnusedImportsCheck)
1. ![CRITICAL][CRITICAL] [Ethereum.java#L28](https://github.com/rsksmart/rskj/blob/e1e0a40121eb30e99d47df989ec1bdeb62520e04/rskj-core/src/main/java/org/ethereum/facade/Ethereum.java#L28): Unused import - org.ethereum.vm.program.ProgramResult. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=checkstyle%3Acom.puppycrawl.tools.checkstyle.checks.imports.UnusedImportsCheck)
1. ![MAJOR][MAJOR] [ReversibleTransactionExecutor.java#L63](https://github.com/rsksmart/rskj/blob/e1e0a40121eb30e99d47df989ec1bdeb62520e04/rskj-core/src/main/java/co/rsk/core/ReversibleTransactionExecutor.java#L63): Method has 8 parameters, which is greater than 7 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00107)
1. ![MAJOR][MAJOR] [RskImpl.java#L39](https://github.com/rsksmart/rskj/blob/e1e0a40121eb30e99d47df989ec1bdeb62520e04/rskj-core/src/main/java/co/rsk/core/RskImpl.java#L39): Constructor has 8 parameters, which is greater than 7 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00107)
1. ![MAJOR][MAJOR] [EthModule.java#L98](https://github.com/rsksmart/rskj/blob/e1e0a40121eb30e99d47df989ec1bdeb62520e04/rskj-core/src/main/java/co/rsk/rpc/modules/eth/EthModule.java#L98): Extract the assignment out of this expression. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AAssignmentInSubExpressionCheck)
1. ![MAJOR][MAJOR] [EthModule.java#L113](https://github.com/rsksmart/rskj/blob/e1e0a40121eb30e99d47df989ec1bdeb62520e04/rskj-core/src/main/java/co/rsk/rpc/modules/eth/EthModule.java#L113): Extract the assignment out of this expression. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AAssignmentInSubExpressionCheck)
1. ![MAJOR][MAJOR] [EthereumImpl.java#L79](https://github.com/rsksmart/rskj/blob/e1e0a40121eb30e99d47df989ec1bdeb62520e04/rskj-core/src/main/java/org/ethereum/facade/EthereumImpl.java#L79): ExecutorService org.ethereum.facade.EthereumImpl.peerServiceExecutor is replaced with another ExecutorService without being shutdown, potentially preventing the entire JVM from shutting down [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3AHES_EXECUTOR_OVERWRITTEN_WITHOUT_SHUTDOWN)
1. ![MAJOR][MAJOR] [EthereumImpl.java#L150](https://github.com/rsksmart/rskj/blob/e1e0a40121eb30e99d47df989ec1bdeb62520e04/rskj-core/src/main/java/org/ethereum/facade/EthereumImpl.java#L150): Non derivable method org.ethereum.facade.EthereumImpl$1.adapt(List) declares throwing an exception that isn't thrown [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ABED_BOGUS_EXCEPTION_DECLARATION)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by rskops** [comment]
SonarQube analysis reported 7 issues

1. ![MAJOR][MAJOR] [ReversibleTransactionExecutor.java#L61](https://github.com/rsksmart/rskj/blob/4f39cb6ddd95766a01889b680e435991cfde507c/rskj-core/src/main/java/co/rsk/core/ReversibleTransactionExecutor.java#L61): Method has 8 parameters, which is greater than 7 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00107)
1. ![MAJOR][MAJOR] [RskImpl.java#L39](https://github.com/rsksmart/rskj/blob/4f39cb6ddd95766a01889b680e435991cfde507c/rskj-core/src/main/java/co/rsk/core/RskImpl.java#L39): Constructor has 8 parameters, which is greater than 7 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00107)
1. ![MAJOR][MAJOR] [EthModule.java#L98](https://github.com/rsksmart/rskj/blob/4f39cb6ddd95766a01889b680e435991cfde507c/rskj-core/src/main/java/co/rsk/rpc/modules/eth/EthModule.java#L98): Extract the assignment out of this expression. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AAssignmentInSubExpressionCheck)
1. ![MAJOR][MAJOR] [EthModule.java#L113](https://github.com/rsksmart/rskj/blob/4f39cb6ddd95766a01889b680e435991cfde507c/rskj-core/src/main/java/co/rsk/rpc/modules/eth/EthModule.java#L113): Extract the assignment out of this expression. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AAssignmentInSubExpressionCheck)
1. ![MAJOR][MAJOR] [EthereumImpl.java#L79](https://github.com/rsksmart/rskj/blob/4f39cb6ddd95766a01889b680e435991cfde507c/rskj-core/src/main/java/org/ethereum/facade/EthereumImpl.java#L79): ExecutorService org.ethereum.facade.EthereumImpl.peerServiceExecutor is replaced with another ExecutorService without being shutdown, potentially preventing the entire JVM from shutting down [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3AHES_EXECUTOR_OVERWRITTEN_WITHOUT_SHUTDOWN)
1. ![MAJOR][MAJOR] [EthereumImpl.java#L150](https://github.com/rsksmart/rskj/blob/4f39cb6ddd95766a01889b680e435991cfde507c/rskj-core/src/main/java/org/ethereum/facade/EthereumImpl.java#L150): Non derivable method org.ethereum.facade.EthereumImpl$1.adapt(List) declares throwing an exception that isn't thrown [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ABED_BOGUS_EXCEPTION_DECLARATION)
1. ![INFO][INFO] [EthModule.java#L45](https://github.com/rsksmart/rskj/blob/4f39cb6ddd95766a01889b680e435991cfde507c/rskj-core/src/main/java/co/rsk/rpc/modules/eth/EthModule.java#L45): Complete the task associated to this TODO comment. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by rskops** [comment]
SonarQube analysis reported 7 issues

1. ![MAJOR][MAJOR] [ReversibleTransactionExecutor.java#L61](https://github.com/rsksmart/rskj/blob/b703d5b80061d1daafcf74b86ca5def1f87692b4/rskj-core/src/main/java/co/rsk/core/ReversibleTransactionExecutor.java#L61): Method has 8 parameters, which is greater than 7 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00107)
1. ![MAJOR][MAJOR] [RskImpl.java#L39](https://github.com/rsksmart/rskj/blob/b703d5b80061d1daafcf74b86ca5def1f87692b4/rskj-core/src/main/java/co/rsk/core/RskImpl.java#L39): Constructor has 8 parameters, which is greater than 7 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00107)
1. ![MAJOR][MAJOR] [EthModule.java#L103](https://github.com/rsksmart/rskj/blob/b703d5b80061d1daafcf74b86ca5def1f87692b4/rskj-core/src/main/java/co/rsk/rpc/modules/eth/EthModule.java#L103): Extract the assignment out of this expression. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AAssignmentInSubExpressionCheck)
1. ![MAJOR][MAJOR] [EthModule.java#L118](https://github.com/rsksmart/rskj/blob/b703d5b80061d1daafcf74b86ca5def1f87692b4/rskj-core/src/main/java/co/rsk/rpc/modules/eth/EthModule.java#L118): Extract the assignment out of this expression. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AAssignmentInSubExpressionCheck)
1. ![MAJOR][MAJOR] [EthereumImpl.java#L79](https://github.com/rsksmart/rskj/blob/b703d5b80061d1daafcf74b86ca5def1f87692b4/rskj-core/src/main/java/org/ethereum/facade/EthereumImpl.java#L79): ExecutorService org.ethereum.facade.EthereumImpl.peerServiceExecutor is replaced with another ExecutorService without being shutdown, potentially preventing the entire JVM from shutting down [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3AHES_EXECUTOR_OVERWRITTEN_WITHOUT_SHUTDOWN)
1. ![MAJOR][MAJOR] [EthereumImpl.java#L150](https://github.com/rsksmart/rskj/blob/b703d5b80061d1daafcf74b86ca5def1f87692b4/rskj-core/src/main/java/org/ethereum/facade/EthereumImpl.java#L150): Non derivable method org.ethereum.facade.EthereumImpl$1.adapt(List) declares throwing an exception that isn't thrown [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ABED_BOGUS_EXCEPTION_DECLARATION)
1. ![INFO][INFO] [EthModule.java#L45](https://github.com/rsksmart/rskj/blob/b703d5b80061d1daafcf74b86ca5def1f87692b4/rskj-core/src/main/java/co/rsk/rpc/modules/eth/EthModule.java#L45): Complete the task associated to this TODO comment. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Pull request merged**